Lib/test/test_builtin.py

```diff
@@ -358,6 +362,43 @@ def f(): """doc"""
                 rv = ns['f']()
                 self.assertEqual(rv, tuple(expected))
 
+    def test_compile_top_level_await(self):
+        """Test whether code some top level await can be compiled.
+
+        Make sure it compiles only with the PyCF_ALLOW_TOP_LEVEL_AWAIT flag set,
+        and make sure the generated code object has the CO_COROUTINE flag set in
+        order to execute it with `yield from eval(.....)` instead of exec.
+        """
+        modes = ('single', 'exec')
+        code_samples = ['''await sleep(0)''',
+        '''async for i in range(10):
+               print(i)''',
+        '''async with asyncio.Lock() as l:
+               pass''']
+        for mode, code_sample in product(modes,code_samples):
+            source = dedent(code_sample)
+            with self.assertRaises(SyntaxError, msg=f"{source=} {mode=}"):
+                compile(source, '?' , mode)
+
+            co = compile(source, '?', mode, flags=ast.PyCF_ALLOW_TOP_LEVEL_AWAIT)
+
+            self.assertEqual(co.co_flags & CO_COROUTINE, CO_COROUTINE,
+                             msg=f"{source=} {mode=}")
```

Comment:
Please add a functional test, that wraps the created code object in a FunctionType and then manually iterates the coroutine (or with asyncio) and checks the result.

Reply:
I've added 2 with `FunctionTypes` and `eval()`